library/src/doo/shell.clj

```diff
@@ -68,7 +75,7 @@
    as it happens by default."
   ([cmd] (sh cmd {:verbose true}))
   ([cmd opts]
-   (let [proc (exec! cmd)
+   (let [proc (exec! cmd (:exec-dir opts))
```

Comment:
I'd probably prefer to see the logic for switching based on whether :exec-dir exists here, rather than further down the call stack. At the moment you have to go three layers deep to see what's happening. OTOH, you're still going to end up with roughly the same code, just a little more verbose. Perhaps you could just add a comment explaining what's happening?